tools/cloud-function-automation/cloud_function_automation.py

```diff
@@ -0,0 +1,278 @@
+    try:
+        publisher.create_topic(name=full_topic_name)
+        print(f"PubSub Topic {full_topic_name} has been created")
+        created_pubsub_topic = True
+    except AlreadyExists:
+        print(f"{full_topic_name} already exists.")
+        created_pubsub_topic = False
+    return created_pubsub_topic
+
+
+def create_sink(sink_name, full_topic_name, log_sink_filter):
+    """
+    This function will be used to create Log Sink from Log Router.
+    :param sink_name: Log Sink name mentioned in variables.py.
+    :param full_topic_name: Pub/Sub topic associated with log sink.
+    :param log_sink_filter: log sink filer mentioned in variables.py.
+    :return: True or False.
+    """
+    try:
+        logging_client = logging.Client()
+    except Exception as exc:
+        raise SystemExit(exc)
+    destination = f"pubsub.googleapis.com/{full_topic_name}"
+    sink = logging_client.sink(sink_name, filter_=log_sink_filter, destination=destination)
+    if sink.exists():
+        print(f"Log Router Sink {sink.name} already exists.")
+        created_sink = False
+    else:
+        try:
+            sink.create()
+        except Exception as exc:
+            raise SystemExit(exc)
+        print(f"Log Router Sink {sink.name} has been created")
+        created_sink = True
+    return created_sink
+
+
+def create_cloud_function(function_config, project_id, full_topic_name):
+    """
+    This function will be used to create and deploy cloud function.
+    :param function_config: dictionary containing cloud function configurations
+    mentioned in variables.py.
+    :param project_id: Google Cloud project id mentioned in variables.py.
+    :param full_topic_name: Pub/Sub topic associated with Cloud function.
+    :return:
+    """
+    function_zip_file_path = function_config["CLOUD_FUNCTION_ZIP_FILE_PATH"]
+    function_location = function_config["CLOUD_FUNCTION_LOCATION"]
+    function_name = function_config["CLOUD_FUNCTION_NAME"]
+    function_runtime = function_config["CLOUD_FUNCTION_RUNTIME"]
+    full_location = f"projects/{project_id}/locations/{function_location}"
+    # Create a client
+    try:
+        client = functions_v1.CloudFunctionsServiceClient()
+    except Exception as exc:
+        raise SystemExit(exc)
+
+    # Initialize request argument(s)
+    function = functions_v1.CloudFunction()
+    function.source_archive_url = function_zip_file_path
+    function.name = f"projects/{project_id}/locations/{function_location}/functions/{function_name}"
+    function.event_trigger = {"event_type": "google.pubsub.topic.publish",
+                              "resource": full_topic_name, "service": "pubsub.googleapis.com"}
+    function.runtime = function_runtime
+
+    request = functions_v1.CreateFunctionRequest(
+        location=full_location,
+        function=function
+    )
+
+    # Make the request
+    try:
+        operation = client.create_function(request=request)
+    except Exception as exc:
```

Comment:
Same here (comments was: What kind of exception can happen here? Handling a generic exception and then exiting immediately does not do much. Please handle the exceptions that you would know how to handle, or at least, that you could enrich with some more meaningful error message.)